### PR TITLE
Update mnist_convnet.py

### PR DESCRIPTION
**Pull request by gawsoftpl**
Fix error change from loss function from categorical_crossentropy to sparse_categorical_crossentropy. Without this example dont run training. For users who just start learn keras that error causes frustration

### CODE REVIEW

**Comment by fchollet** [comment]
This is incorrect, the code uses one-hot encoded labels rather than integer labels, so the loss function must be `categorical_crossentropy`.
